Source/WebCore/css/color/StyleColorLayers.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "CSSPropertyParserConsumer+RawTypes.h"
+#include "ColorInterpolationMethod.h"
+#include "StyleColor.h"
+#include <optional>
+#include <wtf/UniqueRef.h>
+
+namespace WebCore {
+
+enum class BlendMode : uint8_t;
+class Color;
+
+struct StyleColorLayers {
+    WTF_MAKE_STRUCT_FAST_ALLOCATED;
+
+    bool operator==(const StyleColorLayers&) const = default;
+
+    BlendMode blendMode;
+    Vector<StyleColor> colors;
```

Comment:
This can likely be a FixedVector, since it should always be created from a previously known size.

Reply:
Just tried changing it, and it's a bit more of a pain due to the lack of helpers on `FixedVector` (no containsIf method, also `Vector::map` doesn't have a variant that maps into a new FixedVector.

Reply:
(Not impossible to solve these things, but a followup is probably more appropriate given the changes needed in WTF)